src/rgw/rgw_rados.cc

```diff
@@ -733,12 +725,11 @@ int RGWRealm::create(bool exclusive)
     ldout(cct, 0) << "ERROR: failed set current period " << current_period << dendl;
     return ret;
   }
-
-  if (first_realm) { /* this is racy, but it's fine */
-    ret = set_as_default();
-    if (ret < 0) {
-      ldout(cct, 0) << "WARNING: failed to set realm as default realm, ret=" << ret << dendl;
-    }
+  // try to set as default, but pass exclusive=true so we don't override an
+  // existing default
+  ret = set_as_default(true);
```

Comment:
True , but it could be the newer create realm command. It won't be deterministic. It will be the first one to execute set_as_default (or more correctly to execute the put_obj), better from the previous version but still racy